### PR TITLE
Disable Legacy /auction Endpoint By Default

### PR DESCRIPTION
**Pull request by SyntaxNode**
Implements phase 2 of https://github.com/prebid/prebid-server/issues/2011.

### CODE REVIEW

**Review by AlexBVolcy — APPROVED** [review]
LGTM!

File: config/config_test.go [hunk 1]
```diff
@@ -354,7 +354,7 @@ request_validation:
     ipv4_private_networks: ["1.1.1.0/24"]
     ipv6_private_networks: ["1111::/16", "2222::/16"]
 generate_bid_id: true
-enable_legacy_auction: false
+enable_legacy_auction: true
```
Comment (guscarreon) [inline config/config_test.go:357]: For the `TestFullConfig` test, can we leave its value as `false` just for the sake of it being different to the default value of `true`

File: config/config_test.go [hunk 2]
```diff
@@ -579,7 +579,7 @@ func TestFullConfig(t *testing.T) {
 	cmpStrings(t, "request_validation.ipv6_private_networks", cfg.RequestValidation.IPv6PrivateNetworks[1], "2222::/16")
 	cmpBools(t, "generate_bid_id", cfg.GenerateBidID, true)
 	cmpStrings(t, "debug.override_token", cfg.Debug.OverrideToken, "")
-	cmpBools(t, "enable_legacy_auction", cfg.EnableLegacyAuction, false)
+	cmpBools(t, "enable_legacy_auction", cfg.EnableLegacyAuction, true)
```
Comment (guscarreon) [inline config/config_test.go:582]: For the `TestFullConfig` test, can we leave its value as `false` just for the sake of it being different to the default value of `true`?

Comment (guscarreon) [inline config/config_test.go:582]: For the `TestFullConfig` test, can we leave its value as `false` just for the sake of it being different to the default value of `true`?

**Pull request merged**